anime/components/AnimeSearch.vue

```diff
@@ -178,42 +297,201 @@ function previous() {
 	}
 }
 
-function saveClickedAnimeID(id: number) {
+function saveClickedAnimeID(id: number): void {
 	userStore.storeAnimeId(id);
 }
 
-function selectPage(num: number) {
-	userStore.startPageIndex = num * 35 - 35;
-	userStore.endPageIndex = num * 35 + 1;
+function filter(): animeRest[] {
+	const newFilterAnime = [] as animeRest[];
+	const searchResult = [] as animeRest[];
+	userStore.search = text.value;
+
+	if (text.value.length > 0) {
+		userStore.allAnime.filter((anime: animeRest) => {
+			const animeWords = anime.anime_name.toLowerCase().split(" ");
+			if (text.value.slice(-1) == " ") {
+				text.value = text.value.slice(0, -1);
+			}
+			const textResults = text.value.toLocaleLowerCase().split(" ");
+
+			for (let i = 0; i < animeWords.length; i++) {
+				if (animeWords[i].startsWith(textResults[0])) {
+					if (textResults.length == 1) {
+						searchResult.push(anime);
+					} else {
+						const animeWordsSlice = animeWords
+							.slice(i, animeWords.length)
+							.join("")
+							.replace(/[^a-zA-Z ]/, "")
+							.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/, "");
+
+						const textResultsSlice = textResults
+							.join("")
+							.replace(/[^a-zA-Z ]/, "")
+							.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/, "");
+
+						if (animeWordsSlice.startsWith(textResultsSlice)) {
+							searchResult.push(anime);
+						}
+					}
+				}
+			}
+		});
+	} else {
+		searchResult.push(...userStore.allAnime);
+	}
+
+	searchResult.forEach((anime: animeRest) => {
+		const animeGenre = [] as string[];
+		const hasGenre = ref<boolean>(false);
+
+		const animeSeason = ref<string>("");
+		const animeYear = ref<string>("");
+		const animeStatus = anime.status;
+		const animeType = anime.media_type;
+
+		if (anime.season != null) {
+			animeSeason.value = anime.season.split(" ")[0];
+			animeYear.value = anime.season.split(" ")[1];
+		}
+
+		anime.anime_genre.forEach((genre: { genre: string }) => {
+			animeGenre.push(genre.genre);
+		});
+
+		animeGenre.forEach((genre: string) => {
+			if (genre == media_genre.value) {
+				if (!newFilterAnime.includes(anime)) {
+					hasGenre.value = true;
+				}
+			}
+		});
+
+		if (
+			(animeSeason.value == media_season.value || media_season.value == "") &&
+			(animeYear.value == media_year.value || media_year.value == "") &&
+			(animeStatus == media_status.value || media_status.value == "") &&
+			(animeType == media_type.value || media_type.value == "") &&
+			(hasGenre.value || media_genre.value == "")
+		) {
+			if (!newFilterAnime.includes(anime)) {
+				newFilterAnime.push(anime);
+			}
+		}
+	});
+
+	if (media_sort.value == "Release Date") {
```

Comment:
you only need one function for this filter for example you would take in the filter value in and that value will be used for the filter